### PR TITLE
Adjust to breaking changes on https://api.opencovid.ca/

### PR DESCRIPTION
**Pull request by apiology**
This restores health region-level data for Canada and results in `update_prevalence.py` script running successfully to completion.

Fixes https://github.com/microCOVID/microCOVID/issues/1436

### CODE REVIEW

File: update_prevalence.py
Comment (apiology) [inline update_prevalence.py:328]: I removed these because they're not used (we only use the vaccination-related items).  Validation started failing because `total_cases` started returning only null.  Perhaps `api.covid19tracker.ca` had the same issue with the breaking API changes from `api.opencovid.ca`.

File: update_prevalence.py
Comment (apiology) [inline update_prevalence.py:255]: Health region metadata is no longer hosted via an API, but the data is still around in a CSV we can pull.

File: update_prevalence.py
Comment (apiology) [inline update_prevalence.py:292]: Ditto above - now in CSV form

File: update_prevalence.py
Comment (apiology) [inline update_prevalence.py:809]: The spelled-out province/territory name isn't provided in this API anymore, but a mapping file is provided, so this method now takes in the full province/territory name.

File: update_prevalence.py
Comment (apiology) [inline update_prevalence.py:1468]: The current API data gives us the number of people who have gotten one, two, three, four, etc shots.

I don't think it'd be correct to pass that information into this function:

```python
    def get_partially_vaccinated(total_shots, total_fully_vaccinated, shots_for_full_vaccination) -> int:
        return total_shots - shots_for_full_vaccination * total_fully_vaccinated
```

The second parameter is meant to be the total number of shots administered in the area.  Perhaps that's what `cumulative_cvaccine` meant (?), but it's certainly not what `vaccine_coverage_dose_2` means.  I believe that `vaccine_coverage_dose_2` means "the number of people who have gotten two doses of the vaccine.  As such, the logic is simpler:

```python
                total_fully_vaccinated = most_recent.vaccine_coverage_dose_2
                partially_vaccinated = most_recent.vaccine_coverage_dose_1 - most_recent.vaccine_coverage_dose_2
```

File: update_prevalence.py
Comment (jeanpaulrsoucy) [reply update_prevalence.py]: Note that the `pt_names` and `hr_names` parameters allow the PT and HR names to be returned in a variety of formats for both the `timeseries` and `summary` routes.

File: update_prevalence.py
Comment (jeanpaulrsoucy) [reply update_prevalence.py:1525]: Note that both vaccine _coverage_ (% of population) and _administration_ (raw dose numbers) are available. Presently, the former is an adaption PHAC vaccine coverage data (ccodwg/CovidTimelineCanada#21) whereas the latter is an adaption of the PHAC vaccine administration data (ccodwg/CovidTimelineCanada#47). The former `cumulative_cvaccine` value is equivalent to `vaccine_administration_dose_2` in the current dataset.

Comment (apiology) [reply update_prevalence.py:1525]: @jeanpaulrsoucy: Thanks for the clarification!  Is `cumulative_avaccine` the same as `vaccine_administration_dose_1`?  i.e., the number of people who have gotten at least one shot?

The current microCOVID code assumed that `cumulative_avaccine` counts the total number of shots given, not the number of people who have had at least one shot.

Comment (apiology) [reply update_prevalence.py:1525]: @jeanpaulrsoucy : One other question - could you clarify the meaning of `vaccine_administration_total_doses`?  It seems to be the sum of the `vaccine_administration_dose_*` values - but if my assumption is correct about what those fields mean, that wouldn't be a number that would have any meaning.

Comment (jeanpaulrsoucy) [reply update_prevalence.py:1525]: `cumulative_avaccine` is equivalent to `vaccine_administration_total_doses`, which is the sum of all `vaccine_administration_dose_n` values.

File: update_prevalence.py
Comment (apiology) [inline update_prevalence.py:1383]: I would like to do more here to handle more potential API issues - e.g. right now the only cases that would fail would be the CSVs above going away or columns being renamed.  There are multiple additional API calls below this which are currently unhandled.  I think it would confuse the diff more than is worth it for this PR, however, and there are a couple of different approaches to think through, so I'd propose to punt that for now.

File: update_prevalence.py
Comment (apiology) [reply update_prevalence.py:1525]: OK, thanks - I'll dig in on that and double check my assumptions.

Comment (apiology) [reply update_prevalence.py:1525]: All right, I'm convinced that the logic as I added is correct.  The old logic was correct enough until boosters started to happen, at which point the computed number of partially vaccinated people started trending under the real value, as each booster shot administered inappropriately reduced it by one.

I believe the other (existing) use of `get_partially_vaccinated()` is fine, as it uses api.covid19tracker.ca's `total_vaccinations` and `total_vaccinated` fields, which seem like what the function was intended to use.

**Comment by apiology** [comment]
@justinhaaheim @beshaya: I'm awaiting an upstream data fix at https://github.com/andrewthong/covid19tracker-api/issues/118 before this can be merged, but this is otherwise ready for review.  

Here's what's been done to help make sure this is right:

- Copied in example data from the API responses to the Pydantic definitions to help with review to ensure the right fields were being mapped
- A couple of rounds of self-review of the diff
- Ran `./update_prevalence.py` and verified that Canada health-region-level data appears in `location.ts` and that it appears in the UI
- Verified health region names in `.csv` files seem consistent with past health region names.
- Worked through commentary from @jeanpaulrsoucy, owner of the upstream API in question.
- After discovering an inconsistency in population vs total-people-vaccinated numbers, followed up upstream to find issues there, then verified other numbers in Canada are roughly in line with expectations.

I'd welcome suggestions on other ways to QC this.

Sorry for the delay to all who were waiting to get this functionality back!  It's been an interesting learning process getting familiar with microCOVID's data ingestion.

**Comment by apiology** [comment]
Update: https://github.com/andrewthong/covid19tracker-api/issues/118 is resolved!  This is ready to merge once approved.

File: update_prevalence.py
Comment (beshaya) [reply update_prevalence.py:1528]: This seems right to me; previously cumulative_avaccine was all we could get from the API.

**Review by beshaya — APPROVED** [review]

**Comment by beshaya** [comment]
Thanks for the ping! And thanks for the fix!

**Pull request merged**
